ci/e2e-tests.sh

```diff
@@ -386,6 +386,123 @@ EOF
             echo 'Generated config files' >&2
             ;;
 
+        'manual-x509')
+            echo 'Creating self-signed root CA' >&2
+            openssl req \
+                -new \
+                -x509 \
+                -newkey rsa:4096 -keyout device-id-root.key.pem -nodes \
+                -subj "/CN=aziot_root_ca_cert_e2e_test" \
+                -days 30 \
+                -sha256 \
+                -out device-id-root.pem
+
+            echo 'Uploading root CA to IoT Hub...' >&2
+            az iot hub certificate create \
+                --hub-name "$common_resource_name" --name device-id-root \
+                --path device-id-root.pem
+            echo 'Uploaded root CA to IoT Hub' >&2
+
+            echo 'Fetching first etag for verification code request...' >&2
+            etag="$(
+                az iot hub certificate show \
+                --hub-name "$common_resource_name" --name device-id-root \
+                --query etag --output tsv
+            )"
+
+            echo 'Generating verification code and saving new etag...' >&2
+            cloud_certificate="$(
+                az iot hub certificate generate-verification-code \
+                --hub-name "$common_resource_name" --name device-id-root \
+                --etag "$etag"
+            )"
+            etag="$(<<< "$cloud_certificate" jq '.etag' -r)"
+            verification_code="$(
+                <<< "$cloud_certificate" jq '.properties.verificationCode' -r
+            )"
+
+            echo 'Generating CSR for verification cert and signing it with the root CA to get the verification cert.' >&2
+            openssl req \
+                -newkey rsa:2048 -keyout device-id-root-verify.key.pem -nodes \
+                -out device-id-root-verify.csr \
+                -days 1 \
+                -subj "/CN=${verification_code}"
+
+            openssl x509 -req \
+                -in device-id-root-verify.csr \
+                -CA device-id-root.pem -CAkey device-id-root.key.pem \
+                -out device-id-root-verify.pem \
+                -days 365 -CAcreateserial
+
+            echo 'Uploading verification cert to IoT Hub...' >&2
+            az iot hub certificate verify \
+                --hub-name "$common_resource_name" --name device-id-root \
+                --path device-id-root-verify.pem \
+                --etag "$etag"
+            echo 'Uploaded verification cert to IoT Hub' >&2
```

Comment:
There's no point in doing all this (lines 400 to 442).

Reply:
Ah, yes, that's right. The only reason we needed to go through this rigmarole was to verify the root CA for x509_ca, which we aren't using here. Will remove :+1: